File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/SubRecipe.scala
package examples.scala.recipes

import com.ing.baker.recipe.scaladsl.Recipe

object SubRecipe {
  val recipe: Recipe = Recipe("root recipe")
    .withSubRecipe(Recipe("sub recipe 1"))
    .withSubRecipe(Recipe("sub recipe 2")
      .withSubRecipe(Recipe("sub sub recipe 2.1"))
    )
}

File: examples/docs-code-snippets/src/main/java/examples/java/recipes/SubRecipe.java
package examples.java.recipes;

import com.ing.baker.recipe.javadsl.Recipe;

public class SubRecipe {

    public final static Recipe recipe = new Recipe("root recipe")
            .withSubRecipe(new Recipe("sub recipe 1"))
            .withSubRecipe(new Recipe("sub recipe 2")
                    .withSubRecipe(new Recipe("sub sub recipe 2.1"))
            );

}
